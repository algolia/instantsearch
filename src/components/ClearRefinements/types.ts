import { ClearRefinementsRendererOptions } from '../../connectors/clear-refinements/connectClearRefinements';
import {
  ClearRefinementsCSSClasses,
  ClearRefinementsTemplates,
} from '../../widgets/clear-refinements/clear-refinements';

export type ClearRefinementsProps = {
  refine: ClearRefinementsRendererOptions['refine'];
  cssClasses: ClearRefinementsCSSClasses;
  hasRefinements: ClearRefinementsRendererOptions['hasRefinements'];
  templateProps: {
    [key: string]: any;
    templates: ClearRefinementsTemplates;
  };
};
