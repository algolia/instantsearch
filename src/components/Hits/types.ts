import { SearchResults } from 'algoliasearch-helper';
import { BindEventForHits, SendEventForHits } from '../../lib/utils';
import { Hits } from '../../types';
import { HitsCSSClasses, HitsTemplates } from '../../widgets/hits/hits';

export type HitsProps = {
  cssClasses: HitsCSSClasses;
  hits: Hits;
  results: SearchResults;
  sendEvent: SendEventForHits;
  bindEvent: BindEventForHits;
  templateProps: {
    [key: string]: any;
    templates: HitsTemplates;
  };
};
