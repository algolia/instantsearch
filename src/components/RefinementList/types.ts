export type RefinementListProps = {
  facetValues: Array<{
    isRefined: boolean;
    label: string;
    value: string;
  }>;
};
