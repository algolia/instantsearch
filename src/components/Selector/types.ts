export type SelectorProps = {
  cssClasses?: {
    root?: string | string[];
    select?: string | string[];
    option?: string | string[];
  };
  currentValue?: string | number;
  options: Array<{
    value?: string | number;
    label: string;
  }>;
  setValue(value: string): void;
};
