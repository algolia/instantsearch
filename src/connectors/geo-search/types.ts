import { SendEventForHits } from '../../lib/utils';
import {
  Connector,
  GeoLoc,
  Hit,
  TransformItems,
  WidgetRenderState,
} from '../../types';

export type GeoHit = Hit & Required<Pick<Hit, '_geoLoc'>>;

type Bounds = {
  northEast: GeoLoc;
  southWest: GeoLoc;
};

export type GeoSearchRenderState = {
  currentRefinement?: Bounds;
  position?: GeoLoc;
  items: GeoHit[];
  refine(bounds: Bounds): void;
  clearMapRefinement(): void;
  hasMapMoveSinceLastRefine(): boolean;
  isRefineOnMapMove(): boolean;
  isRefinedWithMap(): boolean;
  setMapMoveSinceLastRefine(): void;
  toggleRefineOnMapMove(): void;
  sendEvent: SendEventForHits;
};

export type GeoSearchConnectorParams = {
  enableRefineOnMapMove: boolean;
  transformItems: TransformItems<GeoHit>;
};

export type GeoSearchWidgetDescription = {
  $$type: 'ais.geoSearch';
  renderState: GeoSearchRenderState;
  indexRenderState: {
    geoSearch: WidgetRenderState<
      GeoSearchRenderState,
      GeoSearchConnectorParams
    >;
  };
  indexUiState: {
    geoSearch: {
      /**
       * The rectangular area in geo coordinates.
       * The rectangle is defined by two diagonally opposite points, hence by 4 floats separated by commas.
       *
       * @example '47.3165,4.9665,47.3424,5.0201'
       */
      boundingBox: string;
    };
  };
};

export type GeoSearchConnector = Connector<
  GeoSearchWidgetDescription,
  GeoSearchConnectorParams
>;
