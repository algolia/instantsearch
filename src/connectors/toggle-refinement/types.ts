import type { SendEventForFacet } from '../../lib/utils';
import type { Connector, CreateURL, WidgetRenderState } from '../../types';

type ToggleRefinementConnectorParams = {
  attribute: string;
  on: string | string[];
  off: string | string[];
};

type FacetValue = {
  isRefined: boolean;
  count: number;
};

type ToggleRefinementRenderState = {
  value: {
    name: string;
    isRefined: boolean;
    count: number | null;
    onFacetValue: FacetValue;
    offFacetValue: FacetValue;
  };
  createURL: CreateURL<string>;
  sendEvent: SendEventForFacet;
  canRefine: boolean;
  refine: (value: string) => void;
};

export type ToggleRefinementWidgetDescription = {
  $$type: 'ais.toggleRefinement';
  renderState: ToggleRefinementRenderState;
  indexRenderState: {
    toggleRefinement: {
      [attribute: string]: WidgetRenderState<
        ToggleRefinementRenderState,
        ToggleRefinementConnectorParams
      >;
    };
  };
  indexUiState: {
    toggle: {
      [attribute: string]: boolean;
    };
  };
};

export type ToggleRefinementConnector = Connector<
  ToggleRefinementWidgetDescription,
  ToggleRefinementConnectorParams
>;
