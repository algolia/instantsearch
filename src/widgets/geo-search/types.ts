/* global google EventListener */

import {
  GeoSearchConnectorParams,
  GeoSearchWidgetDescription,
  GeoHit,
} from '../../connectors/geo-search/types';
import { GeoLoc, Template, WidgetFactory } from '../../types';
import { HTMLMarkerArguments } from './createHTMLMarker';

export type GeoSearchTemplates = {
  HTMLMarker: Template<GeoHit>;
  reset: Template;
  toggle: Template;
  redo: Template;
};

export type GeoSearchCSSClasses = {
  /** The root div of the widget. */
  root: string | string[];
  /** The map container of the widget. */
  map: string | string[];
  /** The control element of the widget. */
  control: string | string[];
  /** The label of the control element. */
  label: string | string[];
  /** The selected label of the control element. */
  selectedLabel: string | string[];
  /** The input of the control element. */
  input: string | string[];
  /** The redo search button. */
  redo: string | string[];
  /** The disabled redo search button. */
  disabledRedo: string | string[];
  /** The reset refinement button. */
  reset: string | string[];
};

export type GeoSearchMarker<TOptions> = {
  createOptions?(item: GeoHit): TOptions;
  events: {
    [key: string]: EventListener;
  };
};

export type GeoSearchWidgetParams = {
  initialZoom?: number;
  initialPosition?: GeoLoc;
  templates?: Partial<GeoSearchTemplates>;
  cssClasses?: Partial<GeoSearchCSSClasses>;
  builtInMarker?: GeoSearchMarker<google.maps.MarkerOptions>;
  customHTMLMarker?: GeoSearchMarker<HTMLMarkerArguments>;
  enableRefine?: boolean;
  enableClearMapRefinement?: boolean;
  enableRefineControl?: boolean;
  container: string | HTMLElement;
  googleReference: typeof google;
};

export type GeoSearchWidget = WidgetFactory<
  GeoSearchWidgetDescription & { $$widgetType: 'ais.geoSearch' },
  GeoSearchConnectorParams,
  GeoSearchWidgetParams
>;
