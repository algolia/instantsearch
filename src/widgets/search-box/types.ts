// @TODO: move this to ../search-box.tsx once that exists
import { Template } from '../../types';

export type SearchBoxCSSClasses = {
  root: string;
  form: string;
  input: string;
  submit: string;
  submitIcon: string;
  reset: string;
  resetIcon: string;
  loadingIndicator: string;
  loadingIcon: string;
};

export type SearchBoxTemplates = {
  submit: Template;
  reset: Template;
  loadingIndicator: Template;
};
