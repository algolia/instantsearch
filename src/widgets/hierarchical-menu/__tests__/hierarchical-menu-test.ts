import { render } from 'preact';
import algoliasearchHelper, {
  AlgoliaSearchHelper,
  SearchParameters,
  SearchResults,
} from 'algoliasearch-helper';
import hierarchicalMenu, {
  HierarchicalMenuWidgetParams,
} from '../hierarchical-menu';
import {
  HierarchicalMenuConnectorParams,
  HierarchicalMenuWidgetDescription,
} from '../../../connectors/hierarchical-menu/connectHierarchicalMenu';
import {
  createInitOptions,
  createRenderOptions,
} from '../../../../test/mock/createWidget';
import { createSearchClient } from '../../../../test/mock/createSearchClient';
import { createSingleSearchResponse } from '../../../../test/mock/createAPIResponse';
import { Widget } from '../../../types';

const mockedRender = render as jest.Mock;

jest.mock('preact', () => {
  const module = jest.requireActual('preact');

  module.render = jest.fn();

  return module;
});

describe('hierarchicalMenu()', () => {
  let container: HTMLDivElement;
  let attributes: string[];
  let options: HierarchicalMenuConnectorParams & HierarchicalMenuWidgetParams;
  let widget: Widget<HierarchicalMenuWidgetDescription & {
    widgetParams: HierarchicalMenuConnectorParams;
  }>;

  beforeEach(() => {
    container = document.createElement('div');
    attributes = ['hello', 'world'];
    options = { container, attributes };

    mockedRender.mockClear();
  });

  describe('Usage', () => {
    it('throws without container', () => {
      // @ts-expect-error
      options = { container: undefined, attributes };
      expect(() => hierarchicalMenu(options))
        .toThrowErrorMatchingInlineSnapshot(`
"The \`container\` option is required.

See documentation: https://www.algolia.com/doc/api-reference/widgets/hierarchical-menu/js/"
`);
    });
  });

  describe('render', () => {
    let results: SearchResults<any>;
    let data: SearchResults.HierarchicalFacet;
    let helper: AlgoliaSearchHelper;
    let state: SearchParameters;

    beforeEach(() => {
      data = {
        name: 'baz',
        count: 3,
        path: '',
        isRefined: false,
        data: [
          { name: 'foo', count: 1, path: '', isRefined: false, data: [] },
          { name: 'bar', count: 2, path: '', isRefined: false, data: [] },
        ],
      };
      helper = algoliasearchHelper(createSearchClient(), '');
      helper.toggleFacetRefinement = jest.fn().mockReturnThis();
      helper.search = jest.fn();
      results = new SearchResults(helper.state, [
        createSingleSearchResponse({}),
      ]);
      results.getFacetValues = jest.fn(() => data);
      state = new SearchParameters();
      options = { container, attributes };
    });

    it('understand provided cssClasses', () => {
      const userCssClasses = {
        root: 'root',
        noRefinementRoot: 'noRefinementRoot',
        searchBox: 'searchBox',
        list: 'list',
        childList: 'childList',
        item: 'item',
        selectedItem: 'selectedItem',
        parentItem: 'parentItem',
        link: 'link',
        label: 'label',
        count: 'count',
        noResults: 'noResults',
        showMore: 'showMore',
        disabledShowMore: 'disabledShowMore',
      };
      widget = hierarchicalMenu({ ...options, cssClasses: userCssClasses });

      widget.init!(createInitOptions({ helper }));
      widget.render!(createRenderOptions({ results, state }));

      const [firstRender] = mockedRender.mock.calls;

      expect(firstRender[0].props).toMatchSnapshot();
    });

    it('calls render', () => {
      widget = hierarchicalMenu(options);

      widget.init!(createInitOptions({ helper }));
      widget.render!(createRenderOptions({ results, state }));

      const [firstRender] = mockedRender.mock.calls;

      expect(mockedRender).toHaveBeenCalledTimes(1);
      expect(firstRender[0].props).toMatchSnapshot();
    });

    it('asks for results.getFacetValues', () => {
      widget = hierarchicalMenu(options);

      widget.init!(createInitOptions({ helper }));
      widget.render!(createRenderOptions({ results, state }));

      expect(results.getFacetValues).toHaveBeenCalledTimes(1);
      expect(results.getFacetValues).toHaveBeenCalledWith('hello', {
        sortBy: ['name:asc'],
      });
    });

    it('has a sortBy option', () => {
      widget = hierarchicalMenu({ ...options, sortBy: ['name:asc'] });

      widget.init!(createInitOptions({ helper }));
      widget.render!(createRenderOptions({ results, state }));

      expect(results.getFacetValues).toHaveBeenCalledTimes(1);
      expect(results.getFacetValues).toHaveBeenCalledWith('hello', {
        sortBy: ['name:asc'],
      });
    });

    it('has a templates option', () => {
      widget = hierarchicalMenu({
        ...options,
        templates: {
          item: 'item2',
        },
      });

      widget.init!(createInitOptions({ helper }));
      widget.render!(createRenderOptions({ results, state }));

      const [firstRender] = mockedRender.mock.calls;

      expect(firstRender[0].props).toMatchSnapshot();
    });

    it('has a transformItems options', () => {
      widget = hierarchicalMenu({
        ...options,
        transformItems: items =>
          items.map(item => ({ ...item, transformed: true })),
      });

      widget.init!(createInitOptions({ helper }));
      widget.render!(createRenderOptions({ results, state }));

      const [firstRender] = mockedRender.mock.calls;

      expect(firstRender[0].props).toMatchSnapshot();
    });

    it('sets facetValues to empty array when no results', () => {
      data = {
        name: 'baz',
        count: 0,
        path: '',
        isRefined: false,
        data: [],
      };
      widget = hierarchicalMenu(options);

      widget.init!(createInitOptions({ helper }));
      widget.render!(createRenderOptions({ results, state }));

      const [firstRender] = mockedRender.mock.calls;

      expect(firstRender[0].props).toMatchSnapshot();
    });

    it('has a toggleRefinement method', () => {
      widget = hierarchicalMenu(options);

      widget.init!(createInitOptions({ helper }));
      widget.render!(createRenderOptions({ results, state }));

      const [firstRender] = mockedRender.mock.calls;

      const elementToggleRefinement = firstRender[0].props.toggleRefinement;
      elementToggleRefinement('mom');

      expect(helper.toggleFacetRefinement).toHaveBeenCalledTimes(1);
      expect(helper.toggleFacetRefinement).toHaveBeenCalledWith('hello', 'mom');
      expect(helper.search).toHaveBeenCalledTimes(1);
    });

    it('has a limit option', () => {
      const secondLevel: SearchResults.HierarchicalFacet[] = [
        { name: 'six', path: 'six', count: 6, isRefined: false, data: [] },
        { name: 'seven', path: 'seven', count: 7, isRefined: false, data: [] },
        { name: 'eight', path: 'eight', count: 8, isRefined: false, data: [] },
        { name: 'nine', path: 'nine', count: 9, isRefined: false, data: [] },
      ];
      const firstLevel: SearchResults.HierarchicalFacet[] = [
        { name: 'one', path: 'one', count: 1, isRefined: false, data: [] },
        {
          name: 'two',
          path: 'two',
          count: 2,
          isRefined: false,
          data: secondLevel,
        },
        { name: 'three', path: 'three', count: 3, isRefined: false, data: [] },
        { name: 'four', path: 'four', count: 4, isRefined: false, data: [] },
        { name: 'five', path: 'five', count: 5, isRefined: false, data: [] },
      ];
      data = {
        name: 'zero',
        path: 'zero',
        count: 0,
        isRefined: false,
        data: firstLevel,
      };
      const expectedFacetValues = [
        { label: 'one', value: 'one', count: 1, isRefined: false, data: [] },
        {
          label: 'two',
          value: 'two',
          count: 2,
          isRefined: false,
          data: [
            {
              label: 'six',
              value: 'six',
              count: 6,
              isRefined: false,
              data: [],
            },
            {
              label: 'seven',
              value: 'seven',
              count: 7,
              isRefined: false,
              data: [],
            },
            {
              label: 'eight',
              value: 'eight',
              count: 8,
              isRefined: false,
              data: [],
            },
          ],
        },
        {
          label: 'three',
          value: 'three',
          count: 3,
          isRefined: false,
          data: [],
        },
      ];
      widget = hierarchicalMenu({ ...options, limit: 3 });

      widget.init!(createInitOptions({ helper }));
      widget.render!(createRenderOptions({ results, state }));

      const [firstRender] = mockedRender.mock.calls;

      const actualFacetValues = firstRender[0].props.facetValues;
      expect(actualFacetValues).toEqual(expectedFacetValues);
    });
  });
});
