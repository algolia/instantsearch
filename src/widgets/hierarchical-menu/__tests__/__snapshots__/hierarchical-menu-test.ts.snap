// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hierarchicalMenu() render calls render 1`] = `
{
  "canToggleShowMore": false,
  "createURL": [Function],
  "cssClasses": {
    "childList": "ais-HierarchicalMenu-list--child",
    "count": "ais-HierarchicalMenu-count",
    "disabledShowMore": "ais-HierarchicalMenu-showMore--disabled",
    "item": "ais-HierarchicalMenu-item",
    "label": "ais-HierarchicalMenu-label",
    "link": "ais-HierarchicalMenu-link",
    "list": "ais-HierarchicalMenu-list",
    "noRefinementRoot": "ais-HierarchicalMenu--noRefinement",
    "parentItem": "ais-HierarchicalMenu-item--parent",
    "root": "ais-HierarchicalMenu",
    "selectedItem": "ais-HierarchicalMenu-item--selected",
    "showMore": "ais-HierarchicalMenu-showMore",
  },
  "depth": 0,
  "facetValues": [
    {
      "count": 1,
      "data": [],
      "isRefined": false,
      "label": "foo",
      "value": "",
    },
    {
      "count": 2,
      "data": [],
      "isRefined": false,
      "label": "bar",
      "value": "",
    },
  ],
  "isShowingMore": false,
  "showMore": false,
  "templateProps": {
    "templates": {
      "item": "<a class=\\"{{cssClasses.link}}\\" href=\\"{{url}}\\"><span class=\\"{{cssClasses.label}}\\">{{label}}</span><span class=\\"{{cssClasses.count}}\\">{{#helpers.formatNumber}}{{count}}{{/helpers.formatNumber}}</span></a>",
      "showMoreText": "
    {{#isShowingMore}}
      Show less
    {{/isShowingMore}}
    {{^isShowingMore}}
      Show more
    {{/isShowingMore}}
  ",
    },
    "templatesConfig": {},
    "useCustomCompileOptions": {
      "item": false,
      "showMoreText": false,
    },
  },
  "toggleRefinement": [Function],
  "toggleShowMore": [Function],
}
`;

exports[`hierarchicalMenu() render has a templates option 1`] = `
{
  "canToggleShowMore": false,
  "createURL": [Function],
  "cssClasses": {
    "childList": "ais-HierarchicalMenu-list--child",
    "count": "ais-HierarchicalMenu-count",
    "disabledShowMore": "ais-HierarchicalMenu-showMore--disabled",
    "item": "ais-HierarchicalMenu-item",
    "label": "ais-HierarchicalMenu-label",
    "link": "ais-HierarchicalMenu-link",
    "list": "ais-HierarchicalMenu-list",
    "noRefinementRoot": "ais-HierarchicalMenu--noRefinement",
    "parentItem": "ais-HierarchicalMenu-item--parent",
    "root": "ais-HierarchicalMenu",
    "selectedItem": "ais-HierarchicalMenu-item--selected",
    "showMore": "ais-HierarchicalMenu-showMore",
  },
  "depth": 0,
  "facetValues": [
    {
      "count": 1,
      "data": [],
      "isRefined": false,
      "label": "foo",
      "value": "",
    },
    {
      "count": 2,
      "data": [],
      "isRefined": false,
      "label": "bar",
      "value": "",
    },
  ],
  "isShowingMore": false,
  "showMore": false,
  "templateProps": {
    "templates": {
      "item": "item2",
      "showMoreText": "
    {{#isShowingMore}}
      Show less
    {{/isShowingMore}}
    {{^isShowingMore}}
      Show more
    {{/isShowingMore}}
  ",
    },
    "templatesConfig": {},
    "useCustomCompileOptions": {
      "item": true,
      "showMoreText": false,
    },
  },
  "toggleRefinement": [Function],
  "toggleShowMore": [Function],
}
`;

exports[`hierarchicalMenu() render has a transformItems options 1`] = `
{
  "canToggleShowMore": false,
  "createURL": [Function],
  "cssClasses": {
    "childList": "ais-HierarchicalMenu-list--child",
    "count": "ais-HierarchicalMenu-count",
    "disabledShowMore": "ais-HierarchicalMenu-showMore--disabled",
    "item": "ais-HierarchicalMenu-item",
    "label": "ais-HierarchicalMenu-label",
    "link": "ais-HierarchicalMenu-link",
    "list": "ais-HierarchicalMenu-list",
    "noRefinementRoot": "ais-HierarchicalMenu--noRefinement",
    "parentItem": "ais-HierarchicalMenu-item--parent",
    "root": "ais-HierarchicalMenu",
    "selectedItem": "ais-HierarchicalMenu-item--selected",
    "showMore": "ais-HierarchicalMenu-showMore",
  },
  "depth": 0,
  "facetValues": [
    {
      "count": 1,
      "data": [],
      "isRefined": false,
      "label": "foo",
      "transformed": true,
      "value": "",
    },
    {
      "count": 2,
      "data": [],
      "isRefined": false,
      "label": "bar",
      "transformed": true,
      "value": "",
    },
  ],
  "isShowingMore": false,
  "showMore": false,
  "templateProps": {
    "templates": {
      "item": "<a class=\\"{{cssClasses.link}}\\" href=\\"{{url}}\\"><span class=\\"{{cssClasses.label}}\\">{{label}}</span><span class=\\"{{cssClasses.count}}\\">{{#helpers.formatNumber}}{{count}}{{/helpers.formatNumber}}</span></a>",
      "showMoreText": "
    {{#isShowingMore}}
      Show less
    {{/isShowingMore}}
    {{^isShowingMore}}
      Show more
    {{/isShowingMore}}
  ",
    },
    "templatesConfig": {},
    "useCustomCompileOptions": {
      "item": false,
      "showMoreText": false,
    },
  },
  "toggleRefinement": [Function],
  "toggleShowMore": [Function],
}
`;

exports[`hierarchicalMenu() render sets facetValues to empty array when no results 1`] = `
{
  "canToggleShowMore": false,
  "createURL": [Function],
  "cssClasses": {
    "childList": "ais-HierarchicalMenu-list--child",
    "count": "ais-HierarchicalMenu-count",
    "disabledShowMore": "ais-HierarchicalMenu-showMore--disabled",
    "item": "ais-HierarchicalMenu-item",
    "label": "ais-HierarchicalMenu-label",
    "link": "ais-HierarchicalMenu-link",
    "list": "ais-HierarchicalMenu-list",
    "noRefinementRoot": "ais-HierarchicalMenu--noRefinement",
    "parentItem": "ais-HierarchicalMenu-item--parent",
    "root": "ais-HierarchicalMenu",
    "selectedItem": "ais-HierarchicalMenu-item--selected",
    "showMore": "ais-HierarchicalMenu-showMore",
  },
  "depth": 0,
  "facetValues": [],
  "isShowingMore": false,
  "showMore": false,
  "templateProps": {
    "templates": {
      "item": "<a class=\\"{{cssClasses.link}}\\" href=\\"{{url}}\\"><span class=\\"{{cssClasses.label}}\\">{{label}}</span><span class=\\"{{cssClasses.count}}\\">{{#helpers.formatNumber}}{{count}}{{/helpers.formatNumber}}</span></a>",
      "showMoreText": "
    {{#isShowingMore}}
      Show less
    {{/isShowingMore}}
    {{^isShowingMore}}
      Show more
    {{/isShowingMore}}
  ",
    },
    "templatesConfig": {},
    "useCustomCompileOptions": {
      "item": false,
      "showMoreText": false,
    },
  },
  "toggleRefinement": [Function],
  "toggleShowMore": [Function],
}
`;

exports[`hierarchicalMenu() render understand provided cssClasses 1`] = `
{
  "canToggleShowMore": false,
  "createURL": [Function],
  "cssClasses": {
    "childList": "ais-HierarchicalMenu-list--child childList",
    "count": "ais-HierarchicalMenu-count count",
    "disabledShowMore": "ais-HierarchicalMenu-showMore--disabled disabledShowMore",
    "item": "ais-HierarchicalMenu-item item",
    "label": "ais-HierarchicalMenu-label label",
    "link": "ais-HierarchicalMenu-link link",
    "list": "ais-HierarchicalMenu-list list",
    "noRefinementRoot": "ais-HierarchicalMenu--noRefinement noRefinementRoot",
    "parentItem": "ais-HierarchicalMenu-item--parent parentItem",
    "root": "ais-HierarchicalMenu root",
    "selectedItem": "ais-HierarchicalMenu-item--selected selectedItem",
    "showMore": "ais-HierarchicalMenu-showMore showMore",
  },
  "depth": 0,
  "facetValues": [
    {
      "count": 1,
      "data": [],
      "isRefined": false,
      "label": "foo",
      "value": "",
    },
    {
      "count": 2,
      "data": [],
      "isRefined": false,
      "label": "bar",
      "value": "",
    },
  ],
  "isShowingMore": false,
  "showMore": false,
  "templateProps": {
    "templates": {
      "item": "<a class=\\"{{cssClasses.link}}\\" href=\\"{{url}}\\"><span class=\\"{{cssClasses.label}}\\">{{label}}</span><span class=\\"{{cssClasses.count}}\\">{{#helpers.formatNumber}}{{count}}{{/helpers.formatNumber}}</span></a>",
      "showMoreText": "
    {{#isShowingMore}}
      Show less
    {{/isShowingMore}}
    {{^isShowingMore}}
      Show more
    {{/isShowingMore}}
  ",
    },
    "templatesConfig": {},
    "useCustomCompileOptions": {
      "item": false,
      "showMoreText": false,
    },
  },
  "toggleRefinement": [Function],
  "toggleShowMore": [Function],
}
`;
